src/filters.js

```diff
@@ -191,19 +191,51 @@ var filters = {
             return str.replace(old, new_);
         }
 
-        var res = str;
-        var last = res;
-        var count = 1;
-        res = res.replace(old, new_);
-
-        while(last != res) {
-            if(count >= maxCount) {
-                break;
-            }
+        if(typeof maxCount === "undefined"){
+            maxCount = -1;
+        }
+
+        var res = '';  // Output
+
+        if(typeof old === 'number'){
+            old = old +"";
+        }
+
+        if(typeof str === 'number'){
+            str = str +"";
+        }
+
+        if(typeof old !== "string") {
+            return str;
+        }
+
+        // ShortCircuits
+        if(!old){
```

Comment:
These `if` statements are confusing. Above you say if old is not a string, then return `str`, but here you are treating it like a boolean. If you mean to check for the empty string, please do a strict check like `old !== ''`. Can you combine the above `if` statements into `if/else` where that works too?